.octopus/deployment_process.ocl

```diff
@@ -0,0 +1,192 @@
+                releaseAssets=$(curl -X GET https://api.github.com/repos/$orgName/$repoName/releases | jq "map(select(.name == \"$packageVersion\"))[0].assets")
+                
+                # https://github.com/Homebrew/install/#install-homebrew-on-macos-or-linux
+                NONINTERACTIVE=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+                
+                # https://docs.brew.sh/Manpage#environment
+                HOMEBREW_GIT_EMAIL=$email
+                HOMEBREW_GIT_NAME=$username
+                HOMEBREW_GITHUB_API_TOKEN=$personalAccessToken
+
+                newUrl=$(echo "$releaseAssets" | jq '.[] | .browser_download_url | select(test("macOS_amd64.tar.gz$"; "ig"))')
+                newSha=$(sha256sum "$extractedPath/octopus_$packageVersion_macOS_amd64.tar.gz" | awk '{ print $1 }')
+
+                # https://docs.brew.sh/Manpage#bump-formula-pr-options-formula
+                brew bump-formula-pr \
+                    --fork-org $orgName \
+                    --version $packageVersion \
+                    --url $newUrl \
+                    --sha256 $newSha \
+                    octopus
+            EOT
+            Octopus.Action.Script.ScriptSource = "Inline"
+            Octopus.Action.Script.Syntax = "Bash"
+            OctopusUseBundledTooling = "False"
+        }
+        worker_pool = "hosted-ubuntu"
+
+        container {
+            feed = "docker-hub"
+            image = "octopusdeploy/worker-tools:3.3.2-ubuntu.18.04"
+        }
+
+        packages "cli" {
+            acquisition_location = "Server"
+            feed = "octopus-server-built-in"
+            package_id = "octopus-cli"
+            properties = {
+                Extract = "True"
+                Purpose = ""
+                SelectionMode = "immediate"
+            }
+        }
+    }
+}
+
+step "publish-to-apt-repo" {
+    name = "Publish to APT repo"
+    start_trigger = "StartWithPrevious"
+
+    action {
+        action_type = "Octopus.Script"
+        environments = ["production"]
+        properties = {
+            Octopus.Action.Script.ScriptBody = <<-EOT
+                cd cli || exit
+
+                # Deploy APT repo
+                docker run --rm --volume $(pwd):/working \
+                  --env PUBLISH_LINUX_EXTERNAL="$(get_octopusvariable "Publish:Linux:External")" \
+                  --env PUBLISH_ARTIFACTORY_USERNAME="$(get_octopusvariable "Publish:Artifactory:Username")" \
+                  --env PUBLISH_ARTIFACTORY_PASSWORD="$(get_octopusvariable "Publish:Artifactory:Password")" \
+                  --env AWS_ACCESS_KEY_ID="$(get_octopusvariable "LinuxPackagePublisherAwsAccount.AccessKey")" \
+                  --env AWS_SECRET_ACCESS_KEY="$(get_octopusvariable "LinuxPackagePublisherAwsAccount.SecretKey")" \
+                  octopusdeploy/publish-linux bash -c 'cd /working && bash publish-apt.sh' 2>&1 || exit

```

Comment:
It looks like this docker image is just ubuntu.. but the runner is already ubuntu... what is docker providing for us here?

Reply:
It comes with a couple of extra tools pre-installed that's used by the `publish-*.sh` scripts https://github.com/OctopusDeploy/publish-linux-docker/blob/main/Dockerfile